modules/access_news/access_news.module

```diff
@@ -256,12 +285,258 @@ function affinityGroupBroadcast(EntityInterface $node, $newsType) {
+            ->format(DateTimeItemInterface::DATE_STORAGE_FORMAT);
+
+    $newsHTML = newsRangeView($d1, $d2, $utmCampaign);
+    Drupal::logger('cron_accessnews')->notice('Digest: news from: '. $d1. ' to: '. $d2);
+
+    // gather all upcoming events for the next month (this date to +1 month)
+    $timezone = new \DateTimeZone(DateTimeItemInterface::STORAGE_TIMEZONE);
+    $dtCurrent = new \DateTime('today', $timezone);
+    $dtNextMonth = new \DateTime('today+1 month + 2 day', $timezone);
+
+    $d1 = DrupalDateTime::createFromDateTime($dtCurrent)
+            ->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT);
+    $d2 = DrupalDateTime::createFromDateTime($dtNextMonth)
+            ->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT);
+
+    Drupal::logger('cron_accessnews')->notice('Digest: events from: '. $d1. ' to: '. $d2);
+    $eventHTML = eventsRangeView($d1, $d2, $utmCampaign);
+
+    // put wrapper html
+    $emailHTML = ccNewsRollupHTML($newsHTML, $eventHTML);
+
+    // create constant contact campaign. this does not send it
+    $campaignName = uniqid('Access Weekly Digest: '  .date('n/j/y').' ', FALSE);
+
+    $ccResponse = setupEmailCampaign('The ACCESS Support Digest // ' . date('F j, Y'), $emailHTML, $campaignName);
+
+  } catch (Exception $e) {
+    \Drupal::logger('cron_accessnews')->error('weeklyNewsReport: ' . $e->getMessage());
+  }
+  // todo: if !nosend, send email to xx saying the draft is ready. Or perhaps send a CC test mail to the person.
+    return;
+}
+
+/**
+ * newsRangeView
+ * Return html list of news in this date range.
+ * Each item shows Title, date, time, excerpt, location, link to news item
+ * utmCampaign: text for utm_campaign=
+ */
+function newsRangeView($fromDate, $toDate, $utmCampaign) {
+
+  $listDisplay = '';
+  $divider = dividerHTML();
+
+  try {
+    $nids = \Drupal::entityQuery('node')
+            ->condition('field_published_date.value', $fromDate, '>=')
+            ->condition('field_published_date.value', $toDate, '<=')
+            ->condition('type', 'access_news')
+            ->execute();
+
+    $nodes = Node::loadMultiple($nids);
+    $nCount = 0;
+    foreach ($nodes as $node) {
+
+      try {
+        $nCount += 1;
+
+        $title = $node->getTitle();
+        $pubDate = $node->field_published_date[0]->view('alt_teaser');
+        $pubDateDisp = \Drupal::service('renderer')->renderPlain($pubDate);
+
+        // For body section, meant to be a snippet, this will return either a nicely
+        // truncated version of the body (the length is configured on site via
+        // /admin/structure/types/manage/access_news/display/alt_teaser )
+        // or, if the user has set the "custom summary", will return that instead.
+        // TODO:  The body is returned with <p>body</p> markup, but the summary is not.
+        // (need to distiguish because the news items with and without custom summary
+        //  will have different spacing in the digest.  Or do something with css)
```

Comment:
I think this was dealt with with CSS? Remove the TODO?

Reply:
Sorry, yes, I did fix in css. might be good to keep the note that regular summary and custom summary have this difference., which certainly isn't obvious.